vertical-pod-autoscaler/recommender/recommender.go

```diff
@@ -256,19 +125,26 @@ func createPodResourceRecommender() logic.PodResourceRecommender {
 func NewRecommender(config *rest.Config, metricsFetcherInterval time.Duration, historyProvider history.HistoryProvider) Recommender {
 	recommender := &recommender{
 		clusterState:            model.NewClusterState(),
-		specClient:              newSpecClient(config),
-		metricsClient:           newMetricsClient(config),
+		clusterStateFeeder:      newClusterStateFeeder(config, historyProvider),
 		metricsFetchingInterval: metricsFetcherInterval,
-		historyProvider:         historyProvider,
 		vpaClient:               vpa_clientset.NewForConfigOrDie(config).PocV1alpha1(),
-		vpaLister:               vpa_api_util.NewAllVpasLister(vpa_clientset.NewForConfigOrDie(config), make(chan struct{})),
 		podResourceRecommender:  createPodResourceRecommender(),
 	}
 	glog.V(3).Infof("New Recommender created %+v", recommender)
 
 	return recommender
 }
 
+func newClusterStateFeeder(config *rest.Config, historyProvider history.HistoryProvider) input.ClusterStateFeeder {
```

Comment:
There are two [nN]ewClusterStateFeeder methods. Can you please merge them? Also, how about moving the creation of the clients out from recommender? Ideally, recommender shouldn't have to pull in the input/history, input/metrics, input/spec packages, because it already has clusterFeeder that does everything, right?

Reply:
Thanks for catching the two methods.
I also moved all the clients/lister creation out of recommender, leaving it much skinnier :)